github-connector/internal/application_registry/sendregisterrequest_test.go

```diff
@@ -0,0 +1,107 @@
+package registerservice
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/model"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	exampleID = "123-456789-abcdefghi"
+)
+
+type TestServiceDetails struct {
+	Name string
+}
+
+func TestCreateJSONRequest(t *testing.T) {
+	t.Run("should respond with the same json properties (body, url, method)", func(t *testing.T) {
+		//given
+		JSONBody := TestServiceDetails{
+			Name: "kyma",
+		}
+		requestByte, err := json.Marshal(JSONBody)
+		if err != nil {
+			panic(err.Error)
+		}
+		requestReader := bytes.NewReader(requestByte)
+		config := RequestConfig{
+			Type: "POST",
+			URL:  "http://www.hello-test.com",
+			Body: requestReader,
+		}
+
+		//when
+		req, err := CreateJSONRequest(config)
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(req.Body)
+		s := buf.String()
+
+		//then
+		assert.NoError(t, err)
+		assert.Equal(t, s, string(requestByte))
+		assert.Equal(t, req.URL.String(), config.URL)
+		assert.Equal(t, req.Method, config.Type)
+	})
+}
+
+func StatusOKResponse(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+
+	json.NewEncoder(w).Encode(RegisterResponse{
+		ID: exampleID,
+	})
+}
+func TestSendJSONRequest_TestDataOK(t *testing.T) {
+	t.Run("should response with StatusOK code", func(t *testing.T) {
+		// given
+
+		handler := http.HandlerFunc(StatusOKResponse)
+		server := httptest.NewServer(handler)
+		defer server.Close()
+		req, errNewRequest := http.NewRequest("POST", server.URL, nil)
+		client := server.Client()
+		config := RegisterConfig{
+			HTTPClient:  client,
+			HTTPRequest: req,
+		}
+		//when
+		res, errSendJSON := SendJSONRequest(config)
+		//then
+		assert.Equal(t, res.StatusCode, http.StatusOK)
+		assert.NoError(t, errSendJSON)
```

Comment:
By now this is always passes, because errSendJSON = 'nil'. Changing the error return value in ```SendJSONRequest()``` may (as mentioned above) may be good idea. :)